### PR TITLE
feat(TDQ-19203): Make data preview on the analysis lazy load

### PR DESCRIPTION
**Pull request by zshen-talend**

### CODE REVIEW

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java [hunk 1]
```diff
@@ -203,7 +204,7 @@ private void useCsvReader(File file, DelimitedFileConnection delimitedFileconnec
         try {
             File file = iPath.toFile();
             if (!file.exists()) {
-                return new ArrayList<Object[]>();
+                throw new SQLException("The file is not exist");
```
Comment (qiongli) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:207]: throw an IOException is better

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java [hunk 2]
```diff
@@ -214,8 +215,10 @@ private void useCsvReader(File file, DelimitedFileConnection delimitedFileconnec
             endQuery();
         } catch (IOException e) {
             log.error(e.getMessage(), e);
+            throw new SQLException(e.getMessage(), e);
```
Comment (qiongli) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:218]: throw an IOException is better

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java [hunk 3]
```diff
@@ -214,8 +215,10 @@ private void useCsvReader(File file, DelimitedFileConnection delimitedFileconnec
...
         } catch (Exception e) {
             log.error(e.getMessage(), e);
+            throw new SQLException(e.getMessage(), e);
```
Comment (qiongli) [inline main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:221]: throw an IOException is better

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java (hunk 1 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:207]: Only SQLExcetpion can be throw now. The interface is not permit throw other exception now.

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java (hunk 2 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:218]: Only SQLExcetpion can be throw now. The interface is not permit throw other exception now.

File: main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java (hunk 3 above)
Comment (zshen-talend) [reply main/plugins/org.talend.cwm.management/src/main/java/org/talend/cwm/db/connection/DelimitedFileSQLExecutor.java:221]: Only SQLExcetpion can be throw now. The interface is not permit throw other exception now.

**Review by qiongli — DISMISSED** [review]
After discussion with shenze, keep the SQLException in this commit because it will change many places

**Review by yyin-talend — DISMISSED** [review]

**Review by xqliu-talend — APPROVED** [review]

**Pull request merged**